Update PlayServerEntityEquipment to 1.16 format

Sentropic opened the pull request:
Added methods that work with the new Entity Equipment packet format, using ItemSlot-ItemStack pairs
Marked old methods as deprecated
Corrected wrong descriptions in entity related methods

Sentropic committed:
Update PlayServerEntityEquipment to 1.16 format

Added methods that work with the new Entity Equipment packet format, using ItemSlot-ItemStack pairs
Marked old methods as deprecated

GoryMoon commented on PacketWrapper/src/main/java/com/comphenix/packetwrapper/WrapperPlayServerEntityEquipment.java:
```diff
@@ -61,49 +64,116 @@ public void setEntityID(int value) {
+        public void setSlotStackPair(ItemSlot slot, ItemStack item) {
+            List<Pair<ItemSlot, ItemStack>> slotStackPairs = handle.getSlotStackPairLists().read(0);
+            slotStackPairs.removeIf(pair -> pair.getFirst().equals(slot));
+            slotStackPairs.add(new Pair<>(slot, item));
```
You need to write the slot stack pair list after updating, at least when creating this packet from scratch.
```java
handle.getSlotStackPairLists().write(0, slotStackPairs);
```

dmulloy2 replied:
Same goes for when you're modifying an existing packet

GoryMoon reviewed (changes requested):
When using this and creating a packet from scratch it doesn't set the item slot pair making the client disconnect because of bad packet.
This would need to be added for that to work properly.

JarvisCraft commented:
By the way, there is no need to provide wrappers for `Pair` class itself as it can be made available at compile-time via Mojang's [DataFixerUpper](https://github.com/Mojang/DataFixerUpper) (from which it comes).

dmulloy2 commented on PacketWrapper/src/main/java/com/comphenix/packetwrapper/WrapperPlayServerEntityEquipment.java:
```diff
@@ -61,49 +64,116 @@ public void setEntityID(int value) {
+         * @return true if is set, false otherwise
+         */
+        public boolean isSlotSet(ItemSlot slot) {
+            return handle.getSlotStackPairLists().read(0).stream().anyMatch(pair -> pair.getFirst().equals(slot));
```
Same as above

Sentropic replied:
I suppose here you meant to reference the removeSlotStackPair method? This one doesn't modify the packet's data.

Sentropic committed:
Fixed methods using 1.16+ format

Added line updating packet SlotStackPairLists
Added boolean return value to setSlotStackPair and removeSlotStackPair methods

Sentropic commented:
Still expecting a follow-up on this.
